### DISCUSSION TITLE
Need Help Testing Hardware Acceleration in an unprivileged Jellyfin LXC

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#2857] [Announcements]
**Testing Hardware Acceleration in an unprivileged Jellyfin LXC**
- Default settings are unprivileged
- Privileged LXC is still an option for Hardware Acceleration
- Groups are set automatically
- Checks for the existence of `/dev/dri/card0` if not found use `/dev/dri/card1` and set the GID to `44`
- Sets the GID for `/dev/dri/renderD128` to `104`

```
bash -c "$(wget -qO - https://github.com/tteck/Proxmox/raw/main/ct/jellyfin.sh)"
```
**Please test and share your findings here.** 🙏🏻

**Comment by tteck** [comment]
Or, test Hardware Acceleration in an unprivileged Plex LXC

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/plex.sh)"
```

**Comment by Musrha** [comment]
It works like a charm for me! Using default settings except for putting my apt-cacher-ng IP, after the end of the script the server selection didn't work, but after a LXC reboot the config appeared and did the onboarding just fine.

After login in, I changed Hardware Acceleration to Intel Quicksync and played a movie. In Proxmox VE using intel_gpu_top it show the rendering bar functioning well.

Thank you!!

**Reply by tteck** [reply]
Good to hear! Thank you for the help.

**Comment by EricThunder** [comment]
Running the script yields me an error. I have an NVIDIA card and was using the default script settingsCreating a Jellyfin LXC using the above default settings. (Side note, I didn't have NVIDIA drivers installed, but I don't know what's the best way to install them for everything to work using this new passtrough option)
```
 ✓ Using sabrent for Template Storage.
 ✓ Using sabrent for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 104 was successfully created.
 -Device /dev/dri/renderD128 does not exist

[ERROR] in line 616: exit code 0: while executing command pct start "$CTID"
```

**Reply by tteck** [reply]
Sorry, Intel only. I guess I should state that.

**Reply by pythongod** [reply]
@tteck can you create test tickets for AMD as well?

**Comment by styxadmin** [comment]
I tested Hardware Acceleration in an unprivileged Plex LXC container. Logged into plex and set the Hardware Transcoding device to the UHD Graphics 630 device. 

Transcoding is working perfectly.

![image](https://github.com/tteck/Proxmox/assets/50232595/a143447a-5bfd-4ba0-8ef5-6811a9fc2140)
![image](https://github.com/tteck/Proxmox/assets/50232595/478dc59e-1b83-49b9-b41b-4159e5ee09b5)

**Comment by iaintnodev** [comment]
Not sure if updates broke something (Proxmox VE 8.2.4) or if there's some issue on my side but the HW acceleration in unprivileged Jellyfin container (with default settings) is just not working for me until I use

 `chown :render /dev/dri/renderD128`

as mentioned here https://www.reddit.com/r/Proxmox/comments/19danab/hardware_accelerationtranscoding_not_working_on/.

**Comment by dragonfly8677** [comment]
I'm unable to get it to work on my machine. Running `intel_gpu_top` on my Jellyfin container after a default proxmox-helper-scripts install gives me the following error: `Failed to initialize PMU! (Permission denied)`. Furthermore, activating Intel Quicksync hardware encoding in Jellyfin causes playback to fail.

I checked the group owner of `/dev/dri/renderD128` per the poster above, but it was correctly set to `render` by default.

I'm running a server with an Intel Core i7-10700T which has an Intel UHD Graphics 630 iGPU. Not using any discrete GPU.

**Reply by dragonfly8677** [reply]
@tteck Per the guidelines I won't open a new ticket for this, since this issue is related to this closed ticket. However, it would be appreciated if you could acknowledge that you're aware of this issue.

**Reply by tteck** [reply]
This is not an issue that everyone is having.

**Reply by tteck** [reply]
You cannot utilize `intel_gpu_top` in a unprivileged container.
